pkg/cloud/aws/services/userdata/node.go

```diff
@@ -0,0 +1,66 @@
+// Copyright © 2018 The Kubernetes Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package userdata
+
+const (
+	nodeBashScript = `{{.Header}}
+
+certificate=$(echo '{{.CACert}}' | base64 -w0)
+
+cat >/tmp/cluster-info.yaml <<EOF
+apiVersion: v1
+clusters:
+- cluster:
+    certificate-authority-data: ${certificate}
+    server: https://{{.ELBAddress}}:6443
+  name: ""
+contexts: []
+current-context: ""
+kind: Config
+preferences: {}
+users: []
+EOF
+
+HOSTNAME="$(hostname -f 2>/dev/null || curl http://169.254.169.254/latest/meta-data/local-hostname)"
+
+cat >/tmp/kubeadm-node.yaml <<EOF
+---
+apiVersion: kubeadm.k8s.io/v1alpha3
+kind: NodeConfiguration
+token: {{.BootstrapToken}}
+discoveryFile: /tmp/cluster-info.yaml
+nodeRegistration:
+  name: ${HOSTNAME}
+  criSocket: /var/run/containerd/containerd.sock
+  kubeletExtraArgs:
+    cloud-provider: aws
+EOF
+
+kubeadm join --config /tmp/kubeadm-node.yaml "{{.ELBAddress}}:6443" --ignore-preflight-errors=all
```

Comment:
What preflight errors are we actually ignoring here?

Reply:
Instead of passing in the ELBAddress on the command line it should be set in the config: nodeRegistration.discoveryTokenAPIServers